clients/drcachesim/tracer/raw2trace.h

```diff
@@ -1073,33 +1080,66 @@ template <typename T> class trace_converter_t {
                 (instr->reads_memory() || instr->writes_memory()) &&
                 // No following memref for instruction-only trace type.
                 !is_instr_only_trace) {
-                for (uint j = 0; j < instr->num_mem_srcs(); j++) {
-                    error = append_memref(tls, &buf, instr, instr->mem_src_at(j), false,
-                                          reg_vals);
-                    if (!error.empty())
-                        return error;
-                    error = handle_kernel_interrupt_and_markers(
-                        tls, &buf, cur_modoffs, instr->length(), instrs_are_separate,
-                        &interrupted);
-                    if (!error.empty())
-                        return error;
-                    if (interrupted)
-                        break;
-                }
-                // We break before subsequent memrefs on an interrupt, though with
-                // today's tracer that will never happen (i#3958).
-                for (uint j = 0; !interrupted && j < instr->num_mem_dests(); j++) {
-                    error = append_memref(tls, &buf, instr, instr->mem_dest_at(j), true,
-                                          reg_vals);
-                    if (!error.empty())
-                        return error;
-                    error = handle_kernel_interrupt_and_markers(
-                        tls, &buf, cur_modoffs, instr->length(), instrs_are_separate,
-                        &interrupted);
-                    if (!error.empty())
-                        return error;
-                    if (interrupted)
-                        break;
+                if (instr->is_scatter_or_gather()) {
```

Comment:
This is a large increase in the size of this function.  Could we try one of these:

1) Split something out into a separate function.

2) Instead of this big separate if-else, try to have uniform loops somehow -- not sure it's better.

Reply:
> Split something out into a separate function.

Did this. It did help a bit I think.